src/corehost/cli/deps_format.cpp

```diff
@@ -0,0 +1,259 @@
+        return m_deps_entries[0][index];
+    }
+    return entry;
+}
+
+bool deps_json_t::load_portable(const json_value& json, const pal::string_t& target_name, const rid_fallback_graph_t& rid_fallback_graph)
+{
+    portable_assets_t portable_assets;
+
+    for (const auto& package : json.at(_X("targets")).at(target_name).as_object())
+    {
+        const auto& targets = package.second.as_object();
+        auto iter = targets.find(_X("subtargets"));
+        if (iter == targets.end())
+        {
+            continue;
+        }
+
+        const auto& asset_types = iter->second.as_object();
+        for (int i = 0; i < s_known_asset_types.size(); ++i)
+        {
+            auto iter = asset_types.find(s_known_asset_types[i]);
+            if (iter != asset_types.end())
+            {
+                for (const auto& file : iter->second.as_object())
+                {
+                    const auto& rid = file.second.as_object().at(_X("rid")).as_string();
+                    portable_assets[package.first][rid][i].push_back(file.first);
+                }
+            }
+        }
+    }
+
+    if (!perform_rid_fallback(&portable_assets, rid_fallback_graph))
+    {
+        return false;
+    }
+
+    reconcile_libraries_with_targets(
+        json,
+        [&portable_assets](const pal::string_t& package) -> bool {
+            return portable_assets.find(package) != portable_assets.end();
+        },
+        [&portable_assets](const pal::string_t& package, int type_index) -> const std::vector<pal::string_t>& {
+            return portable_assets[package].begin()->second[type_index];
+        });
+
+    return true;
+}
+
+bool deps_json_t::perform_rid_fallback(portable_assets_t* portable_assets, const rid_fallback_graph_t& rid_fallback_graph)
+{
+    pal::string_t host_rid = get_own_rid();
+    for (auto& package : *portable_assets)
+    {
+        pal::string_t matched_rid = package.second.count(host_rid) ? host_rid : _X("");
+        if (matched_rid.empty())
+        {
+            if (rid_fallback_graph.count(host_rid) == 0)
+            {
+                trace::error(_X("Did not find fallback rids for package %s for the host rid %s"), package.first.c_str(), host_rid.c_str());
+                return false;
+            }
+            const auto& fallback_rids = rid_fallback_graph.find(host_rid)->second;
+            auto iter = std::find_if(fallback_rids.begin(), fallback_rids.end(), [&package](const pal::string_t& rid) {
+                return package.second.count(rid);
+            });
+            if (iter == fallback_rids.end() || (*iter).empty())
+            {
+                trace::error(_X("Did not find a matching fallback rid for package %s for the host rid %s"), package.first.c_str(), host_rid.c_str());
+                return false;
+            }
+            matched_rid = *iter;
+        }
+        assert(!matched_rid.empty());
+        for (auto iter = package.second.begin(); iter != package.second.end(); ++iter)
```

Comment:
Is this just required to cleanup the entries in the package representation once the applicable fallback RID has been found?